pkg/transfer/local/pull.go

```diff
@@ -152,20 +156,24 @@ func (ts *localTransferService) pull(ctx context.Context, ir transfer.ImageFetch
 
 	// TODO: Should available platforms be a configuration of the service?
 	// First find suitable platforms to unpack into
-	//if unpacker, ok := is.
+	//If image storer is also an unpacker type, i.e implemented UnpackPlatforms() func
 	if iu, ok := is.(transfer.ImageUnpacker); ok {
 		unpacks := iu.UnpackPlatforms()
 		if len(unpacks) > 0 {
 			uopts := []unpack.UnpackerOpt{}
 			for _, u := range unpacks {
 				uopts = append(uopts, unpack.WithUnpackPlatform(u))
 			}
-			if ts.limiter != nil {
-				uopts = append(uopts, unpack.WithLimiter(ts.limiter))
+
+			if ts.config.MaxConcurrentDownloads > 0 {
+				limiter = semaphore.NewWeighted(int64(ts.config.MaxConcurrentDownloads))
```

Comment:
This object is intended to be shared to limit the concurrent downloads across all pull operations